toolchain/mfc/run/case_dicts.py

```diff
@@ -20,175 +60,221 @@
+# Feel free to return them if they are needed once more.
 POST_PROCESS = COMMON + [
     't_step_start', 't_step_stop', 't_step_save', 'alt_soundspeed',
     'mixture_err', 'format', 'schlieren_wrt', 'schlieren_alpha', 'fd_order',
-    'fourier_modes%beg', 'fourier_modes%end', 'alpha_rho_wrt', 'rho_wrt',
+    'alpha_rho_wrt', 'rho_wrt',
     'mom_wrt', 'vel_wrt', 'flux_lim', 'flux_wrt', 'E_wrt', 'pres_wrt',
     'alpha_wrt', 'kappa_wrt', 'gamma_wrt', 'heat_ratio_wrt', 'pi_inf_wrt',
     'pres_inf_wrt', 'cons_vars_wrt', 'prim_vars_wrt', 'c_wrt', 'omega_wrt','qbmm',
-    'qm_wrt', 'chem_wrt'
+    'qm_wrt'
 ]
 
 for cmp_id in range(1,3+1):
     cmp = ["x", "y", "z"][cmp_id-1]
 
-    POST_PROCESS.append(f'bc_{cmp}%beg')
-    POST_PROCESS.append(f'bc_{cmp}%end')
+    POST_PROCESS.append(f'bc_{cmp}%beg') # Type already set to int
+    POST_PROCESS.append(f'bc_{cmp}%end') # Type already set to int
 
-    for attribute in ["mom_wrt", "vel_wrt", "flux_wrt", "omega_wrt"]:
-        POST_PROCESS.append(f'{attribute}({cmp_id})')
+    for real_attr in ["mom_wrt", "vel_wrt", "flux_wrt", "omega_wrt"]:
+        add_param(f'{real_attr}({cmp_id})', POST_PROCESS, LOGICALS)
 
-for cmp_id in range(100):
-    POST_PROCESS.append(f'chem_wrt({cmp_id})')
+# NOTE: `chem_wrt` is missing
+# for cmp_id in range(100):
+#     POST_PROCESS.append(f'chem_wrt({cmp_id})')
 
 for fl_id in range(1,10+1):
-    for append in ["schlieren_alpha", "alpha_rho_wrt", "alpha_wrt", "kappa_wrt"]:
-        POST_PROCESS.append(f'{append}({fl_id})')
+    for append, ty in [("schlieren_alpha", REALS), ("alpha_rho_wrt", LOGICALS),
+                       ("alpha_wrt", LOGICALS), ("kappa_wrt", LOGICALS)]:
+        add_param(f'{append}({fl_id})', POST_PROCESS, ty)
 
-    for attribute in ["gamma", "pi_inf", "ss", "pv", "gamma_v", "M_v", "mu_v", "k_v", "G", "mul0",
+    for real_attr in ["gamma", "pi_inf", "ss", "pv", "gamma_v", "M_v", "mu_v", "k_v", "G", "mul0",
                       "cv", "qv", "qvp" ]:
-        POST_PROCESS.append(f"fluid_pp({fl_id})%{attribute}")
+        add_param(f"fluid_pp({fl_id})%{real_attr}", POST_PROCESS, REALS)
 
 ALL = list(set(PRE_PROCESS + SIMULATION + POST_PROCESS))
 
-CASE_OPTIMIZATION = [ "nb", "weno_order", "num_fluids"]
+CASE_OPTIMIZATION = [ "nb", "weno_order", "num_fluids" ]
+
+TYPED_SET = INTEGERS | REALS | LOGICALS | STRINGS
+assert TYPED_SET - set(ALL) == set(), "Found parameter without stage (COMMON, PREPROCESS, SIMULATION, POSTPROCESS)"
+assert set(ALL) - TYPED_SET  == set(), "Found untyped parameter!"
+
+_properties = {}
+for param in ALL:
+    if param in INTEGERS:
+        _properties[param] = {"type": "integer"}
+    elif param in REALS:
+        _properties[param] = {"type": "number"}
+    elif param in LOGICALS:
+        _properties[param] = {"enum": ["T", "F"]}
+    elif param in STRINGS:
+        _properties[param] = {"type": "string"}
+    else:
+        print(f'WARNING: Found parameter {param} without type!')
```

Comment:
I think we should throw an error here instead of just a warning.

Reply:
100% agree